File: src/main/java/ar/edu/um/facultad/rest/model/Persona.java
/**
 * 
 */
package ar.edu.um.facultad.rest.model;

import java.io.Serializable;
import java.math.BigDecimal;

import ar.edu.um.facultad.rest.model.pk.PersonaPk;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.IdClass;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@Entity
@Table
@IdClass(PersonaPk.class)
@NoArgsConstructor
@AllArgsConstructor
public class Persona implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = 6147231331820466299L;

	@Id
	@Column(name = "iddocumento")
	private BigDecimal personaId;

	@Id
	@Column(name = "idtipodoc")
	private Integer documentoId;

	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "id")
	private Long uniqueId;

	private String apellido;
	private String nombre;
	private String sexo;

	@Column(name = "per_pro_id")
	private Integer profesionId;

	@Column(name = "per_mascara")
	private String mascara;

}

File: src/main/java/ar/edu/um/facultad/rest/kotlin/model/Persona.kt
package ar.edu.um.facultad.rest.kotlin.model

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import jakarta.persistence.UniqueConstraint
import java.math.BigDecimal

@Entity
@Table(uniqueConstraints = [UniqueConstraint(columnNames = ["iddocumento", "idtipodoc"])])
data class Persona(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    var uniqueId: Long? = null,

    @Column(name = "iddocumento")
    var personaId: BigDecimal? = null,

    @Column(name = "idtipodoc")
    var documentoId: Int? = null,

    var apellido: String? = null,
    var nombre: String? = null,
    var sexo: String? = null,

    @Column(name = "per_pro_id")
    var profesionId: Int? = null,

    @Column(name = "per_mascara")
    var mascara: String? = null,

    )
